src/arg.c

```diff
@@ -110,6 +110,111 @@ static void dcc_note_compiled(const char *input_file, const char *output_file)
            "compile from %s to %s", input_base, output_base);
 }
 
+/* Look for march/mtune=native in the arguments, and replace with the
+ * actual arguments employed by gcc / clang. */
+static int dcc_resolve_march_native(char* argv[], char*** ret_newargv, int extra_args, int* ignore_range_min, int* ignore_range_max)
+{
+    int i, l, j;
+    l = dcc_argv_len(argv);
+    if (l == 0) return 0;
+    const char* compiler = strrchr(argv[0], '/');
+    compiler = (compiler == 0) ? (argv[0]) : (compiler + 1);
+    int is_clang = strncmp(compiler, "clang", strlen("clang")) == 0;
+    int found_native = 0;
+    for (i = 0;i < l;i++) {
+        if (!strcmp(argv[i], "-march=native") || !strcmp(argv[i], "-mtune=native")) {
+            found_native = 1;
+            break;
+        }
+    }
+    if (!found_native) return 0;
+    char buff[16384];
+    FILE *in;
+    if (!argv[0]) return 0;
+    snprintf(buff, sizeof(buff),
+        "%s -v -E -x c -march=native -mtune=native - < /dev/null 2>&1", argv[0]);
+    if(!(in = popen(buff, "r")))
```

Comment:
Can you please do this without using a shell. I know that is kind of a PITA in C, but adding a dependency on the shell is unacceptable.

Reply:
ok, I can have a look, but actually I have no idea how to do it without a shell

Reply:
https://git.dpkg.org/cgit/dpkg/dpkg.git/tree/dpkg-deb/build.c#n435

Reply:
sorry for the horrible code (and I mangled a bit of your code just to test it easily), but here's one approach using pipe(), dup2 (to replace stdin and stderr for the compiler) and execlp (just because it was easier for testing, not sure if that's easier here, but should be trivial to replace): https://gist.github.com/sandsmark/9f4cc47b6cf4b11f05d677624e0ae025